proposals/stdlib/scan-and-scanReduce.md

```diff
@@ -0,0 +1,208 @@
+
+public inline fun CharSequence.scanReduce(operation: (acc: Char, Char) -> Char): List<Char>
+
+public inline fun ByteArray.scanReduce(operation: (acc: Byte, Byte) -> Byte): List<Byte>
+public inline fun ShortArray.scanReduce(operation: (acc: Short, Short) -> Short): List<Short>
+public inline fun IntArray.scanReduce(operation: (acc: Int, Int) -> Int): List<Int>
+public inline fun LongArray.scanReduce(operation: (acc: Long, Long) -> Long): List<Long>
+public inline fun FloatArray.scanReduce(operation: (acc: Float, Float) -> Float): List<Float>
+public inline fun DoubleArray.scanReduce(operation: (acc: Double, Double) -> Double): List<Double>
+public inline fun BooleanArray.scanReduce(operation: (acc: Boolean, Boolean) -> Boolean): List<Boolean>
+public inline fun CharArray.scanReduce(operation: (acc: Char, Char) -> Char): List<Char>
+
+// indexed variants
+
+public fun <S, T : S> Sequence<T>.scanReduceIndexed(operation: (index: Int, acc: S, T) -> S): Sequence<S>
+
+public inline fun <S, T : S> Iterable<T>.scanReduceIndexed(operation: (index: Int, acc: S, T) -> S): List<S>
+
+public inline fun <S, T : S> Array<out T>.scanReduceIndexed(operation: (index: Int, acc: S, T) -> S): List<S>
+
+public inline fun CharSequence.scanReduceIndexed(operation: (index: Int, acc: Char, Char) -> Char): List<Char>
+
+public inline fun ByteArray.scanReduceIndexed(operation: (index: Int, acc: Byte, Byte) -> Byte): List<Byte>
+public inline fun ShortArray.scanReduceIndexed(operation: (index: Int, acc: Short, Short) -> Short): List<Short>
+public inline fun IntArray.scanReduceIndexed(operation: (index: Int, acc: Int, Int) -> Int): List<Int>
+public inline fun LongArray.scanReduceIndexed(operation: (index: Int, acc: Long, Long) -> Long): List<Long>
+public inline fun FloatArray.scanReduceIndexed(operation: (index: Int, acc: Float, Float) -> Float): List<Float>
+public inline fun DoubleArray.scanReduceIndexed(operation: (index: Int, acc: Double, Double) -> Double): List<Double>
+public inline fun BooleanArray.scanReduceIndexed(operation: (index: Int, acc: Boolean, Boolean) -> Boolean): List<Boolean>
+public inline fun CharArray.scanReduceIndexed(operation: (index: Int, acc: Char, Char) -> Char): List<Char>
+
+public inline fun UIntArray.scanReduceIndexed(operation: (index: Int, acc: UInt, UInt) -> UInt): List<UInt>
+public inline fun ULongArray.scanReduceIndexed(operation: (index: Int, acc: ULong, ULong) -> ULong): List<ULong>
+public inline fun UByteArray.scanReduceIndexed(operation: (index: Int, acc: UByte, UByte) -> UByte): List<UByte>
+public inline fun UShortArray.scanReduceIndexed(operation: (index: Int, acc: UShort, UShort) -> UShort): List<UShort>
+```
+
+## Alternatives
+
+* Use `fold`
+    ```
+    val runningTotal = mutableListOf(0)
+    val total = numbers.fold(0) { acc, element ->
+        val nextAcc = acc + element
+        runningTotal.add(nextAcc)
+        return nextAcc
```

Comment:
return is prohibited here, just `nextAcc`

Reply:
done